batcheval: move ExportRequest to batcheval

dt opened the pull request:
Release justification: very low risk, pure code move for ease of future backports.

Release note: none.

dt committed:
batcheval: move ExportRequest to batcheval

Release justificaiton: very low risk, pure code move for ease of future backports.
Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69295)
<!-- Reviewable:end -->

dt commented on pkg/ccl/backupccl/backup_processor.go:
I'm going to come for this in a follow-up that splits off the setting used by backup to pick its _target_ size from the setting used by kvserver to enforce its _max_ size, so this backup->batcheval edge is temporary

aliher1911 reviewed (approved):
It's happening! 👍

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69295#-:-MhuvrvV1Px7MrtZSv46:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt)



<!-- Sent from Reviewable.io -->

dt committed:
sql: relax settings name lint period rules

The linter was previously spuriously considering any period followed by
a space as implying a sentence, but this is incorrect e.g. when using an
abbreviation like 'e.g.' mid-sentence.

Release note: none.
